common/unsafe/src/main/java/org/apache/spark/unsafe/UnsafeAlignedOffset.java

```diff
@@ -39,7 +39,9 @@ public static int getSize(Object object, long offset) {
       case 8:
         return (int)Platform.getLong(object, offset);
       default:
+        // checkstyle.off: RegexpSinglelineJava
         throw new AssertionError("Illegal UAO_SIZE");
```

Comment:
shall we throw `IllegalStateException` here?

Reply:
I think these are ok as AssertionError because they shouldn't be able to happen in any JVM state

Reply:
It would be OK too. This is just picking nits now but I thought AssertionError was still better. ISE is about application state. Errors are about JVM state.